projectnumber.js

```diff
@@ -1,84 +0,0 @@
-const readline = require('readline');
-const readlineInterface = readline.createInterface(process.stdin, process.stdout);
-
-function ask(questionText) {
-    return new Promise((resolve, reject) => {
-        readlineInterface.question(questionText, resolve);
-    });
-}
-
-async function start() {
-    let maxValue
-    let rangeQuestion= await ask("Would you like to extend the number range?") 
-    
-    if (rangeQuestion==='yes'||rangeQuestion==='yea') {
-        rangeQuestion=await ask ('What range would you like to set?')
-    }
-   
-    var extendedRange=parseInt(rangeQuestion)
-    
-    if ( extendedRange>1) {
-        maxValue= extendedRange
-    }
-    
-    else {
-        maxValue=100
-    }
-
-    let minValue=1
-    let question = await ask ("Let's play a game, are you thinking of a number between 1 -" + maxValue + '?')
-    
-    while (question === 'no') {
-        question = await ask ('Then you should get one please, come back when you do.')
-    }
-    let guessCounter=0
-    var guessArray=[]
-    while (question === 'yes' || question=== 'yea') {
-      //check if valid range min=max=cheat  
-        let guess= Math.floor((maxValue+minValue) / 2)
-       
-        {let first = await ask('Is your number greater, less than , or equal to' + ' ' + guess + '? ')
-        
-        guessArray.push(guess) 
-        if (guessArray===guess){
-            console.log('same guess')
-        }
```

Comment:
Your cheat detector only checks if something has been guessed twice, which wont be the case because the min and max values will be reset. The player should be prompted to input their guess so that you can check against that variable and determine if the player is cheating. Something along the lines of:
`let playerNumber = await ask('What is your number? I wont cheat")`
then you check that number against what the computer guesses, the range values and your guess array.